presto-main/src/main/java/com/facebook/presto/sql/analyzer/Field.java

```diff
@@ -122,7 +123,27 @@ public boolean isAliased()
 
     public boolean matchesPrefix(Optional<QualifiedName> prefix)
     {
-        return !prefix.isPresent() || relationAlias.isPresent() && relationAlias.get().hasSuffix(prefix.get());
+        if (!prefix.isPresent() || relationAlias.isPresent() && relationAlias.get().hasSuffix(prefix.get())) {
+            return true;
+        }
+
+        if (!originTable.isPresent()) {
+            return false;
+        }
+
+        int idx = 0;
+        boolean match = true;
+
+        QualifiedObjectName origin = originTable.get();
+        List<String> prefixes = prefix.get().getParts();
+
+        switch (prefixes.size()) {
```

Comment:
The implicit loop here is unexpected for a switch statement,  which makes it hard to follow and error prone.

Can you change it to a series of if statements instead?
e.g. something like
```
if (prefixes.size = 0) {
    return false;
}

if(prefixes.size() = 3 && !origin.getCatalogName()...) {
     return false;
}
if (prefixes.size() >= 2 && origin.getSchemaName()...) {
     return false;
}
return origin.getObjectName()...;
```

Reply:
Good point! Revised the code to be more explicit! Please see if this looks okay!